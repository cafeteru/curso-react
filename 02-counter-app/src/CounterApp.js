import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 0 }) => {
  const [counter, setCounter] = useState(value);
  const add = () => setCounter(counter + 1);
  const subtract = () => setCounter(counter - 1);
  const reset = () => setCounter(value);
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={add}>+1</button>
      <button onClick={subtract}>-1</button>
      <button onClick={reset}>Reset</button>
    </>
  );
};
CounterApp.propTypes = { value: PropTypes.number };
export default CounterApp;
