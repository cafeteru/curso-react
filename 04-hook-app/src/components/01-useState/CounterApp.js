import { useState } from "react";
import "./counter.css";

export const CounterApp = () => {
  const [state, setState] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });
  const { counter1, counter2 } = state;
  return (
    <>
      <h2>Counter {counter1}</h2>
      <h2>Counter {counter2}</h2>
      <hr></hr>
      <button
        className="btn btn-primary"
        onClick={() =>
          setState({
            ...state,
            counter1: counter1 + 1,
          })
        }
      >
        +1
      </button>
    </>
  );
};
