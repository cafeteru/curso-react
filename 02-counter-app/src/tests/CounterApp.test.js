import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("CounterApp", () => {
  const value = 100;
  let wrapper;
  let buttons;

  beforeEach(() => {
    wrapper = shallow(<CounterApp value={value} />);
    buttons = wrapper.find("button");
    expect(buttons.length).toBe(3);
  });

  test("Debe de mostrar el <CounterApp /> correctamente", () => {
    const wrapper = shallow(<CounterApp />);
    expect(wrapper).toMatchSnapshot();
    const counterValue = wrapper.find("h2").text();
    expect(counterValue).toBe("0");
  });

  test("Debe de mostrar el contador", () => {
    const wrapper = shallow(<CounterApp value={value} />);
    const counterValue = wrapper.find("h2").text();
    expect(counterValue).toBe(value + "");
  });

  test("Debe de incrementar el contador", () => {
    buttons.at(0).simulate("click");
    const counterValue = wrapper.find("h2").text();
    expect(counterValue).toBe(`${value + 1}`);
  });

  test("Debe de decrementar el contador", () => {
    buttons.at(1).simulate("click");
    const counterValue = wrapper.find("h2").text();
    expect(counterValue).toBe(`${value - 1}`);
  });

  test("Debe de resetear el contador", () => {
    buttons.at(1).simulate("click");
    buttons.at(1).simulate("click");
    buttons.at(1).simulate("click");
    buttons.at(2).simulate("click");
    const counterValue = wrapper.find("h2").text();
    expect(counterValue).toBe(`${value}`);
  });
});
