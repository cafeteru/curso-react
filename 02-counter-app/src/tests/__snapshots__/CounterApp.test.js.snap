// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp Debe de mostrar el <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
